actnow/accounts/serializers.py

```diff
@@ -13,7 +13,32 @@ class Meta:
     def create(self, validated_data):
         if not validated_data["username"]:
             del validated_data["username"]
-        user = User.objects.create_user(**validated_data)
+
+        request_data = self.context["request"].data
+
+        first_name = request_data.get("first_name", validated_data["username"])
+        last_name = request_data.get("last_name", "")
+        bio = request_data.get("bio", "")
+        photo = request_data.get("photo", "")
+        location = request_data.get("location", "")
+        phone_number = request_data.get("phone_number", "")
+        social_media_link = request_data.get("social_media_link", "")
+
+        user = User.objects.create_user(
+            **{
+                **validated_data,
+                **{
+                    "first_name": first_name,
+                    "last_name": last_name,
+                    "bio": bio,
+                    "photo": photo,
+                    "location": location,
+                    "phone_number": phone_number,
+                    "social_media_link": social_media_link,
+                },
+            }
+        )
```

Comment:
@esirK I'm abit lost, why are passing `photo`, `bio` & `etc` here. Are we creating a a `UserProfile`?

Reply:
Hey, @KhadijaMahanga  yes. The goal is to pass in both `User Account` details and `User Profile` details and then once we create a User Account we can use the user profile details to create the User Profile.